### ISSUE TITLE
TransformError: ..../lib/index.js: Cannot set property '' of undefined

### ISSUE THREAD

**Issue by buraks** [prescottprue/react-redux-firebase#175]
I am trying to use `react-redux-firebase` with my `react-native` project. I have added `react-redux-firebase@canary` which resolved to `2.0.0-alpha.3`

When I run the app I get:

```
error: bundling: TypeError: Cannot set property '' of undefined
    at Promise$2 (/Users/burak/project/node_modules/es6-promise/dist/lib/es6-promise/promise.js:129:19)
    at Function.memoisePluginContainer (/Users/burak/project/node_modules/babel-core/lib/transformation/file/options/option-manager.js:113:13)
    at Function.normalisePlugin (/Users/burak/project/node_modules/babel-core/lib/transformation/file/options/option-manager.js:146:32)
    at /Users/burak/project/node_modules/babel-core/lib/transformation/file/options/option-manager.js:184:30
    at Array.map (native)
    at Function.normalisePlugins (/Users/burak/project/node_modules/babel-core/lib/transformation/file/options/option-manager.js:158:20)
    at OptionManager.mergeOptions (/Users/burak/project/node_modules/babel-core/lib/transformation/file/options/option-manager.js:234:36)
    at OptionManager.init (/Users/burak/project/node_modules/babel-core/lib/transformation/file/options/option-manager.js:368:12)
    at File.initOptions (/Users/burak/project/node_modules/babel-core/lib/transformation/file/index.js:212:65)
TransformError: /Users/burak/project/node_modules/react-redux-firebase/lib/index.js: Cannot set property '' of undefined
```

I tried to debug the issue. Currently, I only have one import statement below and I am not even calling any method and I get the same error above.

```javascript
import { firebaseStateReducer } from 'react-redux-firebase';
```

Am I missing something?

**Comment by prescottprue** [comment]
@buraks Not sure what the error is from, but I will begin to investigate.

Can you try pointing to `2.0.0-alpha.2` and see if that changes anything for you?

**Comment by buraks** [comment]
Just tried `2.0.0-alpha.2` it's the same

**Comment by prescottprue** [comment]
@buraks What does your babel config look like? Is there any way you can share a repo to reproduce the issue?

**Comment by buraks** [comment]
@prescottprue the project is on a private repo. However, I can share the relevant files with you.

Here is my `package.json`
```javascript
{
	"name": "project",
	"version": "0.0.1",
	"private": true,
	"scripts": {
		"start": "node node_modules/react-native/local-cli/cli.js start",
		"test": "jest"
	},
	"dependencies": {
		"native-base": "2.1.5",
		"react": "16.0.0-alpha.12",
		"react-native": "0.45.1",
		"react-native-code-push": "^3.0.0-beta",
		"react-native-fbsdk": "0.6.0",
		"react-native-firebase": "^2.0.0",
		"react-native-router-flux": "^3.38.0",
		"react-redux": "^5.0.4",
		"react-redux-firebase": "2.0.0-alpha.2",
		"reactotron-react-native": "^1.9.1",
		"reactotron-redux": "^1.9.1",
		"redux": "^3.6.0",
		"redux-persist": "^4.6.0",
		"redux-thunk": "^2.2.0"
	},
	"devDependencies": {
		"babel-jest": "19.0.0",
		"babel-preset-react-native": "1.9.1",
		"eslint": "^3.19.0",
		"eslint-plugin-react": "^6.10.3",
		"jest": "19.0.2",
		"react-test-renderer": "16.0.0-alpha.6"
	},
	"jest": {
		"preset": "react-native"
	}
}
```

Here is the `.babelrc`
```javascript
{
  "presets": ["react-native"]
}
```

Let me know if you need me provide more info. Thanks for taking a look at this issue btw!

**Comment by prescottprue** [comment]
@buraks What version of es6-promise is installed for you (it is a dependency)?

You can find out using `npm ls es6-promise`

**Comment by buraks** [comment]
@prescottprue I use `yarn`. Would that make a change?

```[spotroam{master}]$ yarn list es6-promise
yarn list v0.24.6
└─ es6-promise@4.1.1
✨  Done in 0.79s.```

**Comment by prescottprue** [comment]
So it looks like the issue may be related to that on initial glance. They released a new version a day ago, and the error seams to reference es6-promise.

Kinda a shot in the dark, but could you try switching that to v4.1.0 instead? If this is the case, future versions will have to lock the version of that dependency. Could be totally wrong though.

**Comment by buraks** [comment]
I tried with `es6-promise@4.1.0` same issue 😞 

I also tried running the packager with resetting the cache.
`$ node node_modules/react-native/local-cli/cli.js start --reset-cache` didn't help.

**Comment by buraks** [comment]
@prescottprue here is a gist that includes what changes in my `yarn.lock` when I added `"es6-promise": "4.1.0"` and `"react-redux-firebase": "2.0.0-alpha.2"` to my package.json

https://gist.github.com/buraks/a434357da03493963869f70263327ccf

does anything look suspicious ?

**Comment by projoneftw** [comment]
when initialising with `firebase` or `react-native-firebase`, getting the same issue:
```
UnhandledPromiseRejectionWarning: 
Unhandled promise rejection (rejection id: 1671): SyntaxError: 
TransformError: /...../node_modules/react-redux-firebase/lib/index.js: 
Cannot set property 'oav57qwu3di' of undefined
```
relevant packages: 
```
"react": "~15.4.0-rc.4",
"react-native": "0.42.3",
"react-native-firebase": "^1.0.10",
"react-redux-firebase": "2.0.0-alpha", // tried also 2.0.0-alpha.4
"redux": "^3.6.0",
```

**Comment by prescottprue** [comment]
@projoneftw and @buraks I released [`v2.0.0-alpha.5`](https://github.com/prescottprue/react-redux-firebase/releases/tag/v2.0.0-alpha.5) which removes usage of `es6-promise` all together.

It may not be to blame, but since `es6-promise` was part of the initial error report, I figured it was worth a try.

Still trying to reproduce this error myself to confirm. If either of you get a chance to try it out, let me know. It would be great get this fixed ASAP.

**Comment by projoneftw** [comment]
@prescottprue tried it out, got new error:
```
SyntaxError: TransformError: /.../node_modules/react-redux-firebase/lib/index.js: 
Unknown plugin "add-module-exports" specified in 
"/.../node_modules/react-redux-firebase/.babelrc" at 0, 
attempted to resolve relative to "/.../node_modules/react-redux-firebase"
```

**Comment by prescottprue** [comment]
@projoneftw It seems that this has to do with babelification or transforming of some kind (not sure what would cause it).

Is it possible for you to share a repo where you have this occurring so I can replicate it?

Did you also try the [`react-native` example of the `v2.0.0` branch](https://github.com/prescottprue/react-redux-firebase/tree/v2.0.0)? A few people have been using it already, so I am hoping to find out whats different.

Other random questions (probably unrelated):
* are you running through `react-native run-ios` or a custom command?
* What version iOS/Android?
* Was `v1.5.0-*` working for you? **Note** The [react native syntax is different for `v1.*.*`](http://react-redux-firebase.com/docs/recipes/react-native.html#example-app-snippets)

**Comment by buraks** [comment]
I tried building `react-redux-firebase/examples/complete/react-native` on `v2.0.0-alpha.5` and I got a similar error.

on the packager: `UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 798): Syntax Erro..`

and on the simulator:

![image](https://user-images.githubusercontent.com/2693237/27763101-6c1557ec-5e75-11e7-8147-e056d760857b.png)

**Comment by projoneftw** [comment]
@prescottprue we're working on a private repo... it was originally based on https://github.com/infinitered/ignite boilerplate. 
Answering your questions:

1. I'm using `react-native run-ios`
2. currently iOS 9 (emulator) and Android 6.0.0 (genymotion)
3. haven't tried any version prior to 2.0.0

the error happens in react js packager, so it's the same for ios and droid.
my .babelrc has almost nothing in it (dunno if it's related)
```
{
  "presets": [ "react-native" ],
  "env": {
    "development": {
      "sourceMaps": "inline"
    }
  }
}
```

**Comment by projoneftw** [comment]
@buraks that's exactly what I had before alpha5, u tried "clean run"?
`rm -rf $TMPDIR/react-* && watchman watch-del-all && npm cache clean`
and then `react-native run-ios`

**Comment by prescottprue** [comment]
@projoneftw @buraks and @projoneftw What version of npm and node are you both using?

#179 mentions this same issue, and the reporter seems to think it is due to node_modules not being included. With newer versions of npm, `node_modules` are all flattened and placed on the top level, so I am wondering if the dependencies are loading differently?

**Comment by prescottprue** [comment]
Just released [`v2.0.0-alpha.6`](https://github.com/prescottprue/react-redux-firebase/releases/tag/v2.0.0-alpha.6) that has the `.babelrc` removed and also no longer uses `add-module-exports`.

The hope is that the issue was due to babel looking to the `.babelrc` file inside of the project instead of the main project `.babelrc`.

**Comment by prescottprue** [comment]
Fix was confirmed with @projoneftw and @buraks over gitter.
